### PR TITLE
Allow configuring loopback IP in OutboundIPRangesInclude

### PR DESCRIPTION
**Pull request by shakti-das**
Signed-off-by: Shakti <shaktiprakash.das@salesforce.com>

Fixes: #24142
Allow loopback IP to be specified as input for OutboundIncludeIpRanges (istio-service-cidr)

If loopback IP is explicitly set as an IP for OutboundIncludeIpRanges, then istio-iptables should allow the redirection to envoy_port. This is required to support custom cross-cluster domain suffixes. 

Example:
host: svc.ns.mydomain.test
Where mydomain.test is configured to a loopback IP like 127.1.2.3.
This should reach local envoy (OutboundIncludeIpRanges - 127.1.2.3) so that the VS and DR configured for `svc.ns.mydomain.test` would take effect.

### CODE REVIEW

**Comment by rshriram** [comment]
orthogonal to this PR, could you please give some more explanation for this custom cross domain thing, and its relation to the loopback range? Or put another way, can you (if possible) describe the real scenario (what you are trying to achieve) so that the rest of us get an idea for this kind of change?

**Comment by shakti-das** [comment]
@rshriram In our setup, we have a mix of Baremetal, VMs, and K8s services. The services talk to other services via the service-mesh but also talk to some core services like DB, PKI, etc. without the involvement of the sidecar. So, we always have captured specific outbound IP traffic rather than a wildcard.

All of the variations - Baremetal, VMs, and K8s services use a consistent domain suffix that allows us to convert any address to a specific address that we can capture via sidecar. Like: <svc>.<ns>.mydomain.com. Here, *.mydomain.com would resolve to a loopback IP like 127.1.1.1, and sidecar envoy is configured to capture any outgoing traffic with destination IP as 127.1.1.1.

Once traffic reaches envoy, the `:authority` header match will get us the actual route for the cluster.

This flow breaks with the commit https://github.com/istio/istio/commit/b48ff15218bdf48d4559e94fd18c3d2ee048ae47 because 127.1.1.1 being a loopback IP, it will hit the `-o lo` condition and the App is not running as user 1337. Hence, the packets never reach sidecar's envoy_port.

File: tools/istio-iptables/pkg/cmd/run.go [hunk 1]
```diff
@@ -390,7 +400,10 @@ func (iptConfigurator *IptablesConfigurator) run() {
 
 		// Do not redirect app calls to back itself via Envoy when using the endpoint address
 		// e.g. appN => appN by lo
-		iptConfigurator.iptables.AppendRuleV4(constants.ISTIOOUTPUT, constants.NAT, "-o", "lo", "-m", "owner", "!", "--gid-owner", gid, "-j", constants.RETURN)
+		// If loopback explicitly set via OutboundIPRangesInclude, then don't return.
+		if !ipv4RangesInclude.HasLoopBackIP && !ipv6RangesInclude.HasLoopBackIP {
+			iptConfigurator.iptables.AppendRuleV4(constants.ISTIOOUTPUT, constants.NAT, "-o", "lo", "-m", "owner", "!", "--gid-owner", gid, "-j", constants.RETURN)
+		}
```
Comment (rshriram) [inline tools/istio-iptables/pkg/cmd/run.go:406]: You have to make same change in cni as well.

Comment (shakti-das) [reply tools/istio-iptables/pkg/cmd/run.go:406]: I will send out a PR for CNI.

**Pull request merged**